nexus-repository-conan/src/test/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipeTest.java

```diff
@@ -65,6 +66,31 @@ public void tearDown() {
     System.getProperties().remove(HOSTED_ENABLED_PROPERTY);
   }
 
+  @Test
+  public void canMatchOnConanUpload_url() {
+    when(request.getAction()).thenReturn(POST);
+    when(request.getPath()).thenReturn("/v1/conans/project/2.1.1/group/stable/upload_urls");
+
+    assertTrue(ConanRoutes.uploadUrls().handler(handler).create().getMatcher().matches(context));
+    TokenMatcher.State matcherState = attributesMap.require(TokenMatcher.State.class);
+    assertThat(matcherState.getTokens().get("group"), is(equalTo("group")));
+    assertThat(matcherState.getTokens().get("project"), is(equalTo("project")));
+    assertThat(matcherState.getTokens().get("version"), is(equalTo("2.1.1")));
+  }
+
+  @Test
+  public void canMatchOnConanPackage() {
+    when(request.getAction()).thenReturn(POST);
+    when(request.getPath()).thenReturn("/v1/conans/project/2.1.1/group/stable/packages/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/upload_urls");
+
+    assertTrue(ConanRoutes.uploadUrls().handler(handler).create().getMatcher().matches(context));
+    TokenMatcher.State matcherState = attributesMap.require(TokenMatcher.State.class);
+    assertThat(matcherState.getTokens().get("group"), is(equalTo("group")));
+    assertThat(matcherState.getTokens().get("project"), is(equalTo("project")));
+    assertThat(matcherState.getTokens().get("version"), is(equalTo("2.1.1")));
+    assertThat(matcherState.getTokens().get("sha"), is(equalTo("5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9")));
```

Comment:
What if hash alg will be changed or fields which are used to generate hash will be added/modified/removed

Reply:
Conan client sends requests with prepared hash. This test checks the correct work parser. Answer: nothing, because conan client sends hash and conan plugin saves it